File: resc/WebApplication/src/modules/application/models/contact-person.dto.ts
import { ContactType } from "../enums/contact-type.enum";

export class ContactPerson {
  id: number | null;
  name: string;
  email: string;
  phoneNumber: string;
  type: ContactType;

  constructor(type: ContactType) {
    this.type = type;
  }
}

File: resc/Resc.Application/Applications/Dtos/Create/ContactPersonDto.cs
using Resc.Data.Applications.Enums;

namespace Resc.Application.Applications.Dtos.Create
{
    public class ContactPersonDto
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string PhoneNumber { get; set; }
        public string Email { get; set; }
        public ContactType Type { get; set; }
    }
}
